mmgen/models/diffusions/utils.py

```diff
@@ -0,0 +1,245 @@
+import torch
+
+
+def _get_noise_batch(noise,
+                     image_shape,
+                     num_timesteps=0,
+                     num_batches=0,
+                     timesteps_noise=False):
+    """Get noise batch. Support get sequeue of noise along timesteps.
+    If passed noise is a torch.Tensor,
+        1. timesteps_noise == True
+            1.1. dimension of noise is 4
+                1.1.1. noise shape as [bz, c, h, w]
+                    --> expand noise to [n, bz, c, h, w]
+                1.1.2. noise shape as [n, c, h, w]
+                    --> expand noise to [n, bz, c, h, w]
+                1.1.3. noise shape as [n*bz, c, h, w]
+                    --> view noise to [n, bz, c, h, w]
+            1.2 dimension of noise is 5
+                --> return noise
+        2. timesteps_noise == False
+            2.1 dimension of noise is 3 --> view to [1, c, h, w]
+            2.2 dimension of noise is 4 --> do nothing
+    If passed noise is a callable function or None
+        1. timesteps_noise == True
+            --> generate noise shape as [n, bz, c, h, w]
+        2. timesteps_noise == False
+            --> generate noise shape as [bz, c, h, w]
+    To be noted that, we do not move the generated noise to target device in
```

Comment:
I'm not sure whether this is a standard usage in English. Maybe "It's to be noted that xxx" sounds reasonable.

Reply:
Revised.